docs/source/development/AlgorithmMPISupport.rst

```diff
@@ -266,9 +266,16 @@ In that case the execution mode can simply be determined from the input workspac
 Here the helper ``Parallel::getCorrespondingExecutionMode`` is used to obtain the 'natural' execution mode from a storage mode, i.e., ``ExecutionMode::Identical`` for ``StorageMode::Cloned``, ``ExecutionMode::Distributed`` for ``StorageMode::Distributed``, and ``ExecutionMode::MasterOnly`` for ``StorageMode::MasterOnly``.
```

Comment:
Section on the GUI. Paragraph starting line 394 is not very clear what problem this is solving i.e. what sort of properties would you expect people to add that is MPI related?

Reply:
- ok
- I do not think exposing workspaces indices before MPI was a bad idea. For optimal performance that is even necessary. Likewise, global indices should also be exposed. It is not really possible to "expose" local indices, because whatever the user provides is global by definition (unless you run a different Python script on every MPI rank, such that different indices could be provided). That is what line 366 implies: Any user-provided index must be translated into a local index before using it.
- No, there is only doxygen.
- See for example `CreateWorkspace`. There is no input workspace, so we do not know what execution mode and storage mode to use, without having a property to select it.

Reply:
On the second point. Exposing Indices, Global or otherwise can leave scripts vulnerable as Indices are floating. For example, inserting a `CropWorkspace` at some previous point changes subsequent operations. I can't think of a current workflow example where the one-time translation process would cause issue.

Reply:
I see you point, but the current reality in Mantid seems to be that most algorithms support using workspace indices, and only some of them also support spectrum numbers.